### ISSUE TITLE
Not Usable from within WSL 2 Environment

### ISSUE THREAD

**Issue by Hindol** [Azure/azure-kusto-java#136]
**Describe the bug**
Any attempt to start AAD device code authentication fails with,

```java
1. Caused by java.lang.UnsupportedOperationException
   The BROWSE action is not supported on the current platform!
```

**To Reproduce**

```java
 ConnectionStringBuilder csb = ConnectionStringBuilder.createWithDeviceCodeCredentials(
    "<cluster_uri>"
);
```
And try to run this sample: https://github.com/Azure/azure-kusto-java/blob/master/samples/src/main/java/Query.java

**Expected behavior**
Should be able to recover from a browser not found error in a non-GUI env.

**Desktop (please complete the following information):**
 - OS: Windows 10 / WSL 2
 - Version WSL 2

**Additional context**
More info here: https://bugs.launchpad.net/ubuntu/+source/openjdk-8/+bug/1574879

Basically it boils down to GNOME dependencies not present in a WSL env since WSL is non-GUI.

**Comment by Hindol** [comment]
It seems setting the property `java.awt.headless` to `"true"` can be a temporary workaround.

References: http://developer.classpath.org/doc/java/awt/Desktop-source.html and http://developer.classpath.org/doc/java/awt/GraphicsEnvironment-source.html

**Comment by Hindol** [comment]
Also, to build the project under Java 14, I had to add this snippet to `data/pom.xml` for `maven-javadoc-plugin`.

```xml
<configuration>
    <source>8</source>
</configuration>
```

Reference: https://bugs.openjdk.java.net/browse/JDK-8212233

**Comment by Hindol** [comment]
And the 15 seconds timeout in `AadAuthenticationHelper.java` is way too short for a human to copy the device code from the console, open a browser tab to `https://microsoft.com/devicelogin` and complete two-factor auth (if enabled). A timeout of 30 seconds worked better for me.

**Comment by ohadbitt** [comment]
well it does work on my pc without WSL .
15 seconds is enough if you open the url in advance - then its just a copy paste and maybe 1 other space (or mouse) click - but i agree we can increase to 20. 
what do u suggest for us to do for the java 14 ?

**Comment by ohadbitt** [comment]
btw you can instead do the device authentication on your side and pass the builder the token.

**Comment by Hindol** [comment]
Hi @ohadbitt,

Yeah, I think it should work on vanilla Windows. The issue is, WSL environment is (incorrectly) detected as a desktop environment and the code tries to launch the browser. Arguably, this is a good to have step and not necessary. The library can just print the code to the console and let the user figure out how to open the browser. But in this case I found 15 seconds is too short, given I used my work account and had to do a two-step auth every time.

Regarding Java 14, I made a general comment about what I needed to do to compile this project on my system. Feel free to ignore the comment about Java 14.
